cdap-tms/src/main/java/co/cask/cdap/messaging/store/hbase/HBaseTableFactory.java

```diff
@@ -136,6 +142,88 @@ public PayloadTable createPayloadTable(String tableName) throws IOException {
     );
   }
 
+  public void upgradeMessageTable(String tableName) throws IOException {
+    upgradeCoProcessor(tableUtil.createHTableId(NamespaceId.SYSTEM, tableName),
+                       tableUtil.getMessageTableRegionObserverClassForVersion());
+  }
+
+  public void upgradePayloadTable(String tableName) throws IOException {
+    upgradeCoProcessor(tableUtil.createHTableId(NamespaceId.SYSTEM, tableName),
+                       tableUtil.getPayloadTableRegionObserverClassForVersion());
+  }
+
+  // Disables Message and Payload Table, used in Upgrade Tool
+  public void disableTables(List<String> tableNames) throws IOException {
```

Comment:
Why not follow other methods that take only a `tableName` instead of list? All other methods require the call call it one by one. Is it just for saving the creation of new `HBaseAdmin`?